### ISSUE TITLE
Implement Stable Loss Calculation for Run Tracking

### ISSUE THREAD

**Issue by stepfunction83** [kohya-ss/sd-scripts#1897]
I propose creating a tracker to capture a stable loss measurement as proposed here: https://github.com/spacepxl/demystifying-sd-finetuning

Effectively, at regular intervals during training, a preselected image (or batch of images I imagine) and a preselected noise seed are used to calculate the loss. This ensures that over the course of training, the loss recorded can accurately show the progress of the training run.

By also incorporating a holdout set to use for calculating validation loss, this allows a proper evaluation of what point the model begins to overtrain.

**Comment by stepfunction83** [comment]
It looks like it could be implemented as part of the standard train loop, potentially by creating two additional dataloaders: 

- `test_dataloader` for a test set selected from the dataset itself
- `val_dataloader` for a validation set held out from the dataset and specified separately

It may be additionally helpful to specify seeds for these dataloaders with additional seeds for their respective noise generation processes.

I imagine this could be done with something along the lines of:

`--calc_test_val_loss` (bool, enable test/validation loss calculation)
`--test_dataloader_seed 42`
`--val_dataloader_seed 42`
`--val_dataloader_dir "~/val_dataset_dir/"`

Finally, a frequency should be specified to run loss calculations on these sets:

`--test_val_loss_freq_steps 50` (calculate test/val loss every 50 steps)
`--test_val_loss_freq_epochs 1` (calculate test/val loss every epoch)

**Comment by stepfunction83** [comment]
Implemented in #1899. I am working on a number of enhancements to it in #1900.
